### PR TITLE
Fix the trigger so that it is only invoked when the id is not set

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @bajtos @superkhau

### CODE REVIEW

File: lib/oracle.js [hunk 1]
```diff
@@ -350,7 +350,12 @@ Oracle.prototype.create = function (model, data, callback) {
   var params = fields.params;
   params.push(outParam);
   this.query(sql, params, function (err, info) {
-    if (err) return callback(err);
+    if (err) {
+      if (err && err.toString().indexOf('ORA-00001: unique constraint') >= 0) {
+        err = new Error(err.toString() + '. Duplicate id is detected.');
```
Comment (superkhau) [inline lib/oracle.js]: Minor: remove "is"

File: lib/oracle.js [hunk 2]
```diff
@@ -350,7 +350,12 @@ Oracle.prototype.create = function (model, data, callback) {
   var params = fields.params;
   params.push(outParam);
   this.query(sql, params, function (err, info) {
-    if (err) return callback(err);
+    if (err) {
+      if (err && err.toString().indexOf('ORA-00001: unique constraint') >= 0) {
```
Comment (superkhau) [inline lib/oracle.js]: Double `if (err)...` check here. Maybe:

`if (err) {
  if (err.toString()...
    ...
  return callback(err);
}

Comment (raymondfeng) [inline lib/oracle.js]: Good catch. It was a copy/paste.

**Comment by superkhau** [comment]
Other than the comments, LGTM.

File: lib/oracle.js [hunk 3]
```diff
@@ -350,7 +350,13 @@ Oracle.prototype.create = function (model, data, callback) {
   var params = fields.params;
   params.push(outParam);
   this.query(sql, params, function (err, info) {
-    if (err) return callback(err);
+    if (err) {
+      if (err.toString().indexOf('ORA-00001: unique constraint') >= 0) {
+        // Transform the error so that duplicate can be checked using regex
+        err = new Error(err.toString() + '. Duplicate id detected.');
```
Comment (bajtos) [inline lib/oracle.js]: The error message is misleading unless "ORA-0001: unique constraint" is reported only for primary keys. I think the following text would server better: "Duplicate entry detected."

**Pull request merged**